collector/mdadm_test.go

```diff
@@ -0,0 +1,46 @@
+package collector
+
+import (
+	"testing"
+)
+
+func TestMdadm(t *testing.T) {
+	mdStates, err := parseMdstat("fixtures/mdstat")
+
+	if err != nil {
+		t.FailNow()
+	}
+
+	found := make(map[string]bool)
+	found["md3"] = false
+	found["md127"] = false
+	found["md0"] = false
+	found["md4"] = false
+	found["md6"] = false
+	found["md8"] = false
+	found["md7"] = false
+
+	refs := make(map[string]mdStatus)
+
+	refs["md3"] = mdStatus{"md3",true,8,8,5853468288,5853468288}
+	refs["md127"] = mdStatus{"md127",true,2,2,312319552,312319552}
+	refs["md0"] = mdStatus{"md0",true,2,2,248896,248896}
+	refs["md4"] = mdStatus{"md4",false,2,2,4883648,4883648}
+	refs["md6"] = mdStatus{"md6",true,1,2,195310144,16775552}
+	refs["md8"] = mdStatus{"md8",true,2,2,195310144,16775552}
+	refs["md7"] = mdStatus{"md7",true,3,4,7813735424,7813735424}
```

Comment:
Btw., I'd just do a nested map, so you don't have to repeat `refs` all the time. Like:

```go
refs := map[string]mdStatus{
  "md127": mdStatus{"md127", true, 2, 2, 312319552, 312319552},
  "md0": mdStatus{...},
  // ...
}
```